Makefile

```diff
@@ -5,10 +5,24 @@ IMAGE_NAME=kube-compare
 PACKAGE_NAME          := github.com/openshift/kube-compare
 GOLANG_CROSS_VERSION  ?= v1.22.3
 
+# Default values for GOOS and GOARCH
+GOOS ?= linux
+GOARCH ?= amd64
+
+# These tags make sure we can statically link and avoid shared dependencies
+GO_BUILD_FLAGS :=-tags 'include_gcs include_oss containers_image_openpgp gssapi'
+GO_BUILD_FLAGS_DARWIN :=-tags 'include_gcs include_oss containers_image_openpgp'
+GO_BUILD_FLAGS_WINDOWS :=-tags 'include_gcs include_oss containers_image_openpgp'
+GO_BUILD_FLAGS_LINUX_CROSS :=-tags 'include_gcs include_oss containers_image_openpgp'
+
+OUTPUT_DIR :=_output
+GO_BUILD_BINDIR ?=$(OUTPUT_DIR)/bin
+CROSS_BUILD_BINDIR ?=$(OUTPUT_DIR)/bin
 
 .PHONY: build
 build:
-	go build ./cmd/kubectl-cluster_compare.go
+	mkdir -p $(GO_BUILD_BINDIR)
+	GOOS=$(GOOS) GOARCH=$(GOARCH) go build $(GO_BUILD_FLAGS) -o $(GO_BUILD_BINDIR)/kubectl-cluster_compare ./cmd/kubectl-cluster_compare.go
```

Comment:
Looks good! Do you mind adding a one of two lines with some instruction on how to run this and maybe the what output is? 

Also we are missing a build command for the dockerfile in thie PR? At least example command to build and copy out the binary would nice...and we can update the makefile in the next PR if you would like

Reply:
Good point, I added the image build to the makefile and a comment about the arch types.
Let me know if you think we should add more info there.

Reply:
I guess the bit about "how to copy the binary and save it locally" part in a doc? Maybe something like this (should be the same of podman right? )

```
docker create --name kc kc:v1
docker cp kc:/usr/share/openshift/linux_amd64/kube-compare.rhel9 /local-dir/kube-compare.rhel9
docker rm -f kc
```

But that's probably something we can include in the next PR when we link the upcoming container repo with CI...up to you?

Reply:
Added a basic doc explaining this. I think we adjust it in a different PR if needed

Reply:
Sounds good!